### ISSUE TITLE
excludedFunctions in MagicNumber not working?

### ISSUE THREAD

**Issue by pasniak** [arturbosch/detekt#1936]
Does the following rule should stop reporting magic numbers for functions such as `dip(4)`, `padLeft(4f)` ?  It does not seem to work....

```
  MagicNumber:
    active: true
    excludes: "**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt,**/unused/**,**/*Data.kt"
    excludedFunctions: "pad**,dip**" 
```

**Comment by cortinico** [comment]
Looks like `excludedFunctions` is not a configuration option for `MagicNumber`. Source: https://arturbosch.github.io/detekt/style.html#magicnumber

**Comment by pasniak** [comment]
I use anko to dynamically build Android GUI. GUI widgets are tricky to place and they have to be padded, usually some random padding value which is reported as "MagicNumber". It would be nice to be able to disregard this warning (which I find v useful otherwise).

**Comment by pasniak** [comment]
Also, if it is not a config option I'd suggest a warning or error...

**Comment by cortinico** [comment]
> usually some random padding value which is reported as "MagicNumber"

You can whitelist specific values (like 8/16/32) to do not be reported by this rule.

> Also, if it is not a config option I'd suggest a warning or error...

Agree, but that goes on config file validation, that is another problem :)

**Comment by arturbosch** [comment]
> > Also, if it is not a config option I'd suggest a warning or error...
> 
> Agree, but that goes on config file validation, that is another problem :)

There is #516 which we plan for 1.2.0.

Hm this brings the idea of a new extension possibility: `FindingsListener` or `SuppressionProcessor` to filter findings based on some criteria.
